### PR TITLE
feat: Enable user to get personal schedule added to calendar

### LINKED ISSUE #6694: My Schedule: Enable user to get personal schedule added to calendar
Instead of the overall calendar users should have the option to only save their personal "my schedule" to their calendar. We already have the my schedule feature upcoming. Please add the option to "add it to the calendar".

### PR DESCRIPTION
**Pull request by ranjitkshah**
Fixes #6694 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
My Schedule should include sessions, just event doesn't make any sense

**Comment by ranjitkshah** [comment]
> My Schedule should include sessions, just event doesn't make any sense

yes yes I know that's why i drafted

**Comment by ranjitkshah** [comment]
> > My Schedule should include sessions, just event doesn't make any sense
> 
> yes yes I know that's why i drafted

I passed as `.ics?include_sessions&my_schedule`  but Its not working. its added all session not a my schedule

**Comment by iamareebjamal** [comment]
Try now

**Comment by iamareebjamal** [comment]
Just realized this won't work

**Comment by iamareebjamal** [comment]
Server PR is merged, please complete this

**Comment by ranjitkshah** [comment]
> Server PR is merged, please complete this

still not working

File: app/components/public/add-to-calender.ts [hunk 1]
```diff
@@ -101,12 +101,12 @@ export default class AddToCalender extends Component<Args> {
...
   get mySessionGoogleUrl(): string {
     const { event } = this.args;
-    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule';
+    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&user_id';
   }
 
   get mySessioniCalUrl(): string {
     const host = this.loader.host();
-    return host + '/v1/events/' + this.args.event.identifier + '.ics?include_sessions&my_schedule';
+    return host + '/v1/events/' + this.args.event.identifier + '.ics?include_sessions&my_schedule&user_id';
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:109]: What difference does it make?

Comment (ranjitkshah) [reply app/components/public/add-to-calender.ts:109]: confuse here, what should I pass as user_id?  where can I get user_id

Comment (iamareebjamal) [reply app/components/public/add-to-calender.ts:109]: authManager.currentUser

File: app/components/public/add-to-calender.ts [hunk 2]
```diff
@@ -101,12 +105,12 @@ export default class AddToCalender extends Component<Args> {
 
   get mySessionGoogleUrl(): string {
     const { event } = this.args;
-    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&user_id';
+    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&' + this.authManager.currentUser.email;
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:108]: What?

File: app/components/public/add-to-calender.ts [hunk 3]
```diff
@@ -101,12 +105,12 @@ export default class AddToCalender extends Component<Args> {
...
   }
 
   get mySessioniCalUrl(): string {
     const host = this.loader.host();
-    return host + '/v1/events/' + this.args.event.identifier + '.ics?include_sessions&my_schedule&user_id';
+    return host + '/v1/events/' + this.args.event.identifier + '.ics?include_sessions&my_schedule&' + this.authManager.currentUser.email;
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:113]: What?

File: app/components/public/add-to-calender.ts [hunk 4]
```diff
@@ -105,12 +105,12 @@ export default class AddToCalender extends Component<Args> {
 
   get mySessionGoogleUrl(): string {
     const { event } = this.args;
-    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&' + this.authManager.currentUser.email;
+    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&' + this.authManager.currentUser.id;
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:108]: What?

File: app/components/public/add-to-calender.ts [hunk 5]
```diff
@@ -105,12 +105,12 @@ export default class AddToCalender extends Component<Args> {
...
   }
 
   get mySessioniCalUrl(): string {
     const host = this.loader.host();
-    return host + '/v1/events/' + this.args.event.identifier + '.ics?include_sessions&my_schedule&' + this.authManager.currentUser.email;
+    return host + '/v1/events/' + this.args.event.identifier + '.ics?include_sessions&my_schedule&' + this.authManager.currentUser.id;
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:113]: What?

**Comment by iamareebjamal** [comment]
Are you even testing your changes locally before pushing?

**Comment by ranjitkshah** [comment]
> Are you even testing your changes locally before pushing?

```
 get mySessionGoogleUrl(): string {
    const { event } = this.args;
    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&user_id=' + this.authManager.currentUser.id;
  }
```

using this but it adds all the sessions ☹️

**Comment by iamareebjamal** [comment]
No issue, just push the changes

**Comment by iamareebjamal** [comment]
Showing My Schedule to logged out users as well

File: app/components/public/add-to-calender.ts [hunk 6]
```diff
@@ -99,6 +103,16 @@ export default class AddToCalender extends Component<Args> {
     return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions';
   }
 
+  get mySessionGoogleUrl(): string {
+    const { event } = this.args;
+    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + '.ics?include_sessions&my_schedule&user_id=' + this.authManager.currentUser.id;
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:108]:
```suggestion
    return 'https://calendar.google.com/calendar/render?cid=webcal://api.eventyay.com/v1/events/' + event.identifier + encodeURIComponent('.ics?include_sessions&my_schedule&user_id=') + this.authManager.currentUser.id;
```

File: app/components/public/add-to-calender.hbs [hunk 7]
```diff
@@ -14,6 +14,7 @@
         <button {{action 'openAddToCalendarModal' this.calendarUrls}} class="ui item inverted fluid button">{{t 'Add Event'}}</button>
         {{#if (and @event.isSchedulePublished this.showSessions)}}
           <button {{action 'openAddToCalendarModal' this.sessionCalendarUrls}} class="ui item inverted fluid button">{{t 'Add Sessions'}}</button>
+          <button {{action 'openAddToCalendarModal' this.mySessionCalendarUrls}} class="ui item inverted fluid button">{{t 'Add My Schedule'}}</button>
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:17]:
```suggestion
          {{#if this.session.isAuthenticated}}
            <button {{action 'openAddToCalendarModal' this.mySessionCalendarUrls}} class="ui item inverted fluid button">{{t 'Add My Schedule'}}</button>
          {{/if}}
```

File: app/components/public/add-to-calender.hbs [hunk 8]
```diff
@@ -14,6 +14,9 @@
         <button {{action 'openAddToCalendarModal' this.calendarUrls}} class="ui item inverted fluid button">{{t 'Add Event'}}</button>
         {{#if (and @event.isSchedulePublished this.showSessions)}}
           <button {{action 'openAddToCalendarModal' this.sessionCalendarUrls}} class="ui item inverted fluid button">{{t 'Add Sessions'}}</button>
+          {{#if this.session.isAuthenticated}}
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:17]:
```suggestion
          {{#if this.authManager.currentUser}}
```

**Comment by ranjitkshah** [comment]
I am also changing this.session.authenticated to authmanager thanks it's working now

**Pull request merged**